src/client/Microsoft.Identity.Client/Internal/Requests/RequestBase.cs

```diff
@@ -96,7 +96,9 @@ public async Task<AuthenticationResult> RunAsync(CancellationToken cancellationT
 
                         authenticationResult.AuthenticationResultMetadata.DurationTotalInMs = sw.ElapsedMilliseconds;
                         authenticationResult.AuthenticationResultMetadata.DurationInHttpInMs = apiEvent.DurationInHttpInMs;
-                        authenticationResult.AuthenticationResultMetadata.DurationInCacheInMs = apiEvent.DurationInCacheInMs;                        
+                        authenticationResult.AuthenticationResultMetadata.DurationInCacheInMs = apiEvent.DurationInCacheInMs;
+
+                        authenticationResult.AuthenticationResultMetadata.AzureRegionAutoDetected = Region.RegionManager.AutoDiscoveredRegion;
```

Comment:
question to discuss w/team: we had talked about doing a helper method which would return the region, so developers could call that upfront and use it directly, or call it and see what it returns. is there value in having it come after the fact in the auth result, or are we still considering the helper method? or both?

Reply:
Additional thoughts. Since region auto discovery is done once, then cached, region in auth result will always be the same. So maybe a helper method or a public static property on like RegionManager might be more appropriate. Also the discovery is done when AcquireTokenX is called, so before that first call, region helper will return null, unless we add discovery logic to it (which I think we also discussed). I think the primary question is what is the end users' scenario for needing a region here.

Reply:
@jmprieur thoughts on this?
I think @pmaytak found the right question:

> what is the end users' scenario for needing a region here.

Reply:
@pmaytak The end user scenario for needing this is to confirm the region that is being used to hit the regional endpoint. And I remember we discussed this and decided that we should expose the auto detected region but when I think again, I think we should actually expose the region used instead. @bgavrilMS to confirm

Reply:
static string ConfidentialClientApplication.AutoDetecteRegion() ?

Reply:
@neha-bhargava - if the user configures a region, it is always used. Is there a way to detect if the region was ok or not? I don't think there is.